### PR TITLE
expose spec for dynamically generated messages

### PR DESCRIPTION
**Pull request by dirk-thomas**
Required for ros/ros_comm#1006.

@ros/ros_team Please review.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder retest this please

File: src/genpy/dynamic.py [hunk 1]
```diff
@@ -190,6 +190,7 @@ def generate_dynamic(core_type, msg_cat):
             messages[t] = getattr(mod, _gen_dyn_name(pkg, s_type))
         except AttributeError:
             raise MsgGenerationException("cannot retrieve message class for %s/%s: %s"%(pkg, s_type, _gen_dyn_name(pkg, s_type)))
+        messages[t]._spec = specs[t]
```
Comment (dirk-thomas) [inline src/genpy/dynamic.py]: The instance is a generated class. It doesn't have that attribute before. `_full_text` contains the plain text of the message, `_spec` contains the parsed [MsgSpec](https://github.com/ros/genmsg/blob/490b3753213ed3e2251acd218c62224b2787ee0e/src/genmsg/msgs.py#L228) instance for the message.

Comment (mikaelarguedas) [inline src/genpy/dynamic.py]: ok that explains what I couldn't find it in the generated code then. Thanks for clarifying :+1:

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder retest this please

**Comment by dirk-thomas** [comment]
@ros/ros_team Please review.

**Comment by wjwwood** [comment]
Isn't @mikaelarguedas already reviewing it?

**Comment by dirk-thomas** [comment]
I am still waiting for an approval on this.

**Pull request merged**